src/AudioIO.cpp

```diff
@@ -1554,6 +1552,12 @@ void AudioIO::StopStream()
       }
    }
 
+
+   // No longer need effects processing. This must be done after the stream is stopped
+   // to prevent the callback from being invoked after the effects are finalized.
+   mpTransportState.reset();
```

Comment:
This line moves very much later, so you must ask whether reodering it with any of the things it passed over has any important consequences.  I think not.  Still, I would not move it so very late.

Reply:
I suggest:  split the `if (mStreamToken > 0) {` block, that is close it earlier, then test the condition again.  In between, destroy state.

The block `if (mPlaybackTracks.size() > 0)` is harmless if it is outside the `if (mStreamToken > 0) {` block, and now destroying of state is next to it.

And that is good:  now the cleanup steps, duplicated here and in `AudioIO::StartStreamCleanup`, are next to resetting of mpTransportState in both places.

That means the other cleanup I would like to see -- moving the repeated steps into the destructor of mpTransportState, eliminating duplication -- can be done as a next step in a later PR.

Reply:
The elimination. of duplicate code can be a later pull request.

Do the things I suggested though, and I'm not yet satisfied.

Use stronger memory ordering, acquire and release, for `mAudioThreadShouldCallTrackBufferExchangeOnce` before I consider this fix correct.  But maybe the other atomic variables are not needed yet.

Reply:
Alright, suggestions implemented in next commit